### PR TITLE
WHFHRI-671

### PR DESCRIPTION
**Pull request by alisakotliarova**
Migrate Travis builds to GitHub Actions

## PR Requirements:
 - [ ] The code must be reviewed by at least one Maintainer
 - [x] JIRA Ticket: https://jira.wh-sdlc.watson-health.ibm.com/browse/WHFHRI-671
 - [x] 90 % test code coverage
 - [x] README updates

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -25,21 +25,21 @@ ext {
     slf4jVersion = '1.7.7'
     log4jVersion = '1.2.17'
     jacksonVersion = '2.12.0'
-    branch = System.getenv('TRAVIS_BRANCH') != null
-        ? System.getenv('TRAVIS_BRANCH')
+    branch = System.getenv('ACTIONS_BRANCH') != null
```
Comment (FRicci618) [inline build.gradle:28]: Where is `ACTIONS_BRANCH` defined? I'd recommend using the `nelonoel/branch-name@v1.0.1` action, like in `hri-mgmt-api`. Then you can call `BRANCH_NAME`.

File: build.gradle [hunk 2]
```diff
@@ -92,7 +89,7 @@ dependencies {
     // Dependencies that library users should include in their job
     // shadow jar
     // --------------------------------------------------------------
-    implementation "org.alvearie.hri:hri-api-batch-notification:develop-SNAPSHOT"
+    implementation "org.alvearie.hri:hri-api-batch-notification:WHFHRI-693-SNAPSHOT"
```
Comment (FRicci618) [inline build.gradle:92]: Don't forget to switch this back before merging.

File: .github/workflows/ci-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+name: CI Workflow
+
+on:
+  [push, workflow_dispatch]
```
Comment (dan-baxter) [inline .github/workflows/ci-workflow.yml:4]: This looks different than what I'd seen and copied from mgmt-api: 

```

on: 

  push:

    branches: [ '**' ]

  workflow_dispatch:

```

It's possible they behave the same, but I'm not totally sure how this syntax works. Do you know/did you do it this way for another reason?

File: .github/workflows/ci-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+concurrency: ci-${{ github.ref }}
+
+jobs:
+  buildAndTest:
+    name: Build, Test, and Package
+    runs-on: ubuntu-latest
+    env:
+      RESOURCE_GROUP: hri-dev1-wdc-kube
```
Comment (dan-baxter) [inline .github/workflows/ci-workflow.yml:13]: Nit -- a newline after the env vars would be good

File: .github/workflows/ci-workflow.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          curl -sL https://ibm.biz/idt-installer | bash
+          ibmcloud plugin install doi
+
+      - name: Log into IBM Cloud
+        shell: bash
+        env:
+          CLOUD_API_KEY: ${{ secrets.CLOUD_API_KEY }}
+        run: |
+          IBMCLOUD_API_KEY="$CLOUD_API_KEY" ibmcloud login -r us-south || { echo 'IBM Cloud CLI login failed!'; exit 1; }
+          ibmcloud target -g ${RESOURCE_GROUP}
```
Comment (dan-baxter) [inline .github/workflows/ci-workflow.yml:28]: I don't think we actually have to do the two above steps (Install IBM Dev Tools & DevOps Insights CLI plug-in and Log into IBM Cloud) anymore. I'm guessing that was only needed either for the previous "run-insights-publish" script or for publishing to artifactory -- either way I think they can go.

File: build.gradle [hunk 6]
```diff
@@ -111,27 +108,24 @@ dependencies {
...
 publishing {
     publications {
-        mavenJava(MavenPublication) {
-            from components.java
-            artifactId = 'hri-flink-pipeline-core'
-        }
-        mavenJava(MavenPublication) {
-            artifact jarTests
+        gpr(MavenPublication) {
+            from(components.java)
```
Comment (dan-baxter) [inline build.gradle:118]: I'm not sure I understand this change, could you explain?

File: build.gradle [hunk 7]
```diff
@@ -25,21 +25,21 @@ ext {
     slf4jVersion = '1.7.7'
     log4jVersion = '1.2.17'
     jacksonVersion = '2.12.0'
-    branch = System.getenv('TRAVIS_BRANCH') != null
-        ? System.getenv('TRAVIS_BRANCH')
+    branch = System.getenv('ACTIONS_BRANCH') != null
+        ? System.getenv('ACTIONS_BRANCH')
```
Comment (aramcodz) [inline build.gradle:29]: Similar to Fred's comment. 

Q. Team,  have we decided on a standardized name change for what we had been calling `TRAVIS_BRANCH`??

I have seen us use Both `ACTIONS_BRANCH` (as @alisakotliarova has here) and `BRANCH_NAME`. 

We should pick one version and stick with it.

Thoughts, @FRicci618?

File: .github/workflows/ci-workflow.yml (hunk 4 above)
Comment (aramcodz) [reply .github/workflows/ci-workflow.yml:13]: +1 for Dan's Comment. Agree.

File: .github/workflows/ci-workflow.yml (hunk 5 above)
Comment (aramcodz) [reply .github/workflows/ci-workflow.yml:28]: I believe @dan-baxter is correct. Any objections to removing this?

**Review by aramcodz — COMMENTED** [review]
Couple of comments.

File: build.gradle (hunk 6 above)
Comment (aramcodz) [reply build.gradle:118]: Yeah, I was wondering this too.  What is `gpr`?

File: .github/workflows/ci-workflow.yml (hunk 3 above)
Comment (alisakotliarova) [reply .github/workflows/ci-workflow.yml:4]: I replaced that syntax from what I originally had (looked more like `hri-mgmt-api`) based off of the [ci-workflow.yaml](https://github.com/Alvearie/hri-api-spec/blob/develop/.github/workflows/ci-workflow.yaml) in `hri-api-spec`.  I don't mind using the `hri-mgmt-api` version, I'll go ahead and replace it.

File: .github/workflows/ci-workflow.yml (hunk 5 above)
Comment (alisakotliarova) [reply .github/workflows/ci-workflow.yml]: I removed the section and the build still passes.

File: build.gradle (hunk 1 above)
Comment (alisakotliarova) [reply build.gradle]: I replaced it with the action ✅

**Review by aramcodz — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (alisakotliarova) [reply build.gradle]: Switched ✅
For some reason when I was working on this before, it wasn't cooperating with the `develop-SNAPSHOT`.  But I switched it back and it's passing now.

File: build.gradle [hunk 8]
```diff
@@ -111,27 +107,24 @@ dependencies {
     testImplementation "org.apache.flink:flink-streaming-java_${scalaBinaryVersion}:${flinkVersion}:tests"
 
     testRuntimeOnly "com.vladsch.flexmark:flexmark-all:0.35.10"
-    testRuntime 'org.pegdown:pegdown:1.4.2'
-    testCompile "info.picocli:picocli:4.2.0"
+    testRuntimeOnly 'org.pegdown:pegdown:1.4.2'
+    testImplementation "info.picocli:picocli:4.2.0"
```
Comment (David-N-Perkins) [inline build.gradle:111]: Why did you need to change these?

**Review by David-N-Perkins — COMMENTED** [review]
Why was `grandlew` changed? Did you upgrade the Gradle version?

File: build.gradle (hunk 8 above)
Comment (alisakotliarova) [reply build.gradle]: The previous variables were deprecated and these are the new names of them.

File: build.gradle [hunk 9]
```diff
@@ -56,14 +55,11 @@ tasks.withType(JavaCompile) {
...
-                username = findProperty('user') ?: System.getenv('user')
-                password = findProperty('password') ?: System.getenv('password')
-            }
-
-            url "TBD/$repo/"
+    maven {
+        url = uri("https://maven.pkg.github.com/Alvearie/hri-api-spec")
+        credentials {
+            username = System.getenv('USER')
+            password = System.getenv('PASSWORD')
```
[the remark this answers is not in the document]
Comment (alisakotliarova) [reply build.gradle]: I've added that back in
